kits/daisy/src/main.cpp

```diff
@@ -0,0 +1,648 @@
+#include <QGuiApplication>
+#include <QGraphicsScene>
+#include <QGraphicsView>
+#include <QMessageBox>
+
+#include <QtWidgets/QApplication>
+#include <QtWidgets/QWidget>
+#include <QtWidgets/QLabel>
+#include <QtWidgets/QHBoxLayout>
+
+#include "display/hexapod_view_2d.hpp"
+
+#include "robot/hexapod.hpp"
+#include "input/input_manager_mobile_io.hpp"
+#include <atomic>
+#include <iostream>
+#include <unistd.h>
+#include <chrono>
+#include <thread>
+#include <set>
+
+using namespace hebi;
+using namespace Eigen;
+
+bool parse_parameters(int argc, char** argv, bool& visualize, bool& dummy, bool& partial, bool& quiet, std::set<int>& partial_legs)
+{
+  visualize = false;
+  dummy = false;
+  partial = false;
+  quiet = false;
+  bool valid = true;
+  int idx = 1; // Ignore program name!
+  for (;idx < argc; ++idx)
+  {
+    std::string str_arg(argv[idx]);
+    if (str_arg == "-h")
+    {
+      std::cout << "Hexapod control usage:" << std::endl <<
+      "    -d" << std::endl <<
+      "        Dummy hexapod; no modules are connected to.  Incompatible with \"-p\"." << std::endl << std::endl <<
+      "    -p <list of integer leg indices>" << std::endl <<
+      "        Creates a partial dummy hexapod; only the legs with the specified indices" << std::endl <<
+      "        are connected to." << std::endl << std::endl <<
+      "    -v" << std::endl <<
+      "        Visualize -- show a simple rendering of the robot." << std::endl << std::endl <<
+      "    -q" << std::endl <<
+      "        Quiet mode (no dialog messages; waits and tries to continue on failure such as no modules on the network)." << std::endl <<
+      "    -h" << std::endl <<
+      "        Print this help and return." << std::endl;
+      return false;
+    }
+    else if (str_arg == "-v")
+    {
+      visualize = true;
+      continue;
+    }
+    else if (str_arg == "-d")
+    {
+      dummy = true;
+      continue;
+    }
+    else if (str_arg == "-p" && idx + 1 < argc)
+    {
+      partial = true;
+      for (++idx; idx < argc; ++idx)
+      {
+        try
+        {
+          int partial_leg = std::stoi(std::string(argv[idx]));
+          partial_legs.insert(partial_leg);
+        }
+        catch (const std::invalid_argument&)
+        {
+          idx -= 2; // -2, because we will add one next start of the parent loop
```

Comment:
How much have you tested this logic?

Reply:
quite a bit

Reply:
Still not pretty; it might be worth using a header-only argument parsing library in place of this -- we had one in the old C API examples that I liked.

Reply:
Good enough for me